provisioning/image_cache.go

```diff
@@ -187,6 +187,11 @@ func newImageCachePodTemplateSpec(info *ProvisioningInfo) (*corev1.PodTemplateSp
 		},
 	}
 
+	nodeSelector := map[string]string{}
+	if !info.HypershiftEnabled {
```

Comment:
Image cache is a legacy. It was needed when we still supported qcow2-based provisioning, and it is still needed for environments upgraded from before 4.10. It is never needed on HCP, so let's make sure we don't run it on HCP at all (see EnsureImageCache in this file).

Reply:
Done